packages/financial-templates-lib/src/price-feed/USPACPriceFeed.ts

```diff
@@ -0,0 +1,214 @@
+import assert = require("assert");
+import { FixedNumber } from "@ethersproject/bignumber";
+import { PriceFeedInterface } from "./PriceFeedInterface";
+import moment from "moment-timezone";
+import type { Logger } from "winston";
+import Web3 from "web3";
+import { BN } from "../types";
+import { NetworkerInterface } from "./Networker";
+
+interface MarketstackResponse {
+  data?: { date: string; open: number; close: number; symbol: string }[];
+}
+
+export class USPACPriceFeed extends PriceFeedInterface {
+  private readonly uuid: string;
+  private readonly toBN = Web3.utils.toBN;
+  private currentPrice: BN | null = null;
+  private lastUpdateTime: number | null = null;
+
+  /**
+   * @notice Constructs the USPACPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {String[]} symbols Array of symbols that constitute basket
+   * @param {String} marketstackApiKey Marketstack API access key.
+   * @param {boolean} useHttps If use https for Marketstack (only available on paid plans)
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {Number} priceFeedDecimals Number of priceFeedDecimals to use to convert price to wei.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   */
+  constructor(
+    private readonly logger: Logger,
+    private readonly web3: Web3,
+    private readonly symbols: string[],
+    private readonly correctionFactor: string,
+    private readonly marketstackApiKey: string,
+    private readonly useHttps: boolean,
+    private readonly lookback: number,
+    private readonly networker: NetworkerInterface,
+    private readonly getTime: () => Promise<number>,
+    private readonly priceFeedDecimals = 4 /* Prices in Marketstack are 2 decimals, but give some extra precision */,
```

Comment:
`priceFeedDecimals` parameter has no relation to input precision, but how to format returned price. It was important before UMA contracts could deal with different collateral decimals, but now this should default to 18

Reply:
Fixed